src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,181 @@
+                                caseNumber, dailyListHearing.getHearingDetails().getHearingDate()
+                        );
+
+                        CourtCaseEntity courtCase = hearing.getCourtCase();
+                        addJudges(sitting, hearing);
+                        addDefendants(courtCase, dailyListHearing.getDefendants());
+                        addProsecutions(courtCase, dailyListHearing.getProsecution().getAdvocate().getPersonalDetails().getName());
+                        addDefenders(courtCase, dailyListHearing.getDefendants());
+
+                        courtCase.addHearing(hearing);
+
+                        hearingRepository.saveAndFlush(hearing);
+                        caseRepository.saveAndFlush(courtCase);
+                    }
+                }
+            } else {
+                statusType = JobStatusType.PARTIALLY_PROCESSED;
+                log.error("Unregistered courthouse " + courtHouseName + " daily list entry has not been processed");
+            }
+        }
+        dailyListEntity.setStatus(statusType.name());
+    }
+
+    private String getCaseNumber(List<Defendant> defendants, DailyListEntity dailyListEntity, Hearing hearing) {
+        // CPP don't provide case id, use URN
+        if (String.valueOf(SourceType.CPP).equalsIgnoreCase(dailyListEntity.getSource())) {
+            if (defendants.isEmpty()) {
+                return hearing.getCaseNumber();
+            } else {
+                String urn = defendants.get(0).getUrn();
+                if (StringUtils.isEmpty(urn)) {
+                    log.error("Hearing not added - HearingInfo does not contain a URN value");
+                } else {
+                    return urn;
+                }
+            }
+        }
+        return hearing.getCaseNumber();
+
+    }
+
+
+    private void addProsecutions(CourtCaseEntity courtCase, CitizenName prosecutionName) {
+        courtCase.addProsecutor(prosecutorRepository.createProsecutor(buildFullName(prosecutionName), courtCase));
+    }
+
+
+    private void addDefenders(CourtCaseEntity courtCase, List<Defendant> defendants) {
+        for (Defendant defendant : defendants) {
+            courtCase.addDefence(defenceRepository.createDefence(
+                    buildFullName(defendant.getCounsel().getAdvocate().getPersonalDetails().getName()), courtCase));
+        }
+    }
+
+    private void addDefendants(CourtCaseEntity courtCase, List<Defendant> defendants) {
+        for (Defendant defendant : defendants) {
+            courtCase.addDefendant(defendantRepository.createDefendant(buildFullName(defendant.getPersonalDetails().getName()), courtCase));
+        }
+
+    }
+
+    private void addJudges(Sitting sitting, HearingEntity hearing) {
+        for (CitizenName judge : sitting.getJudiciary()) {
+            JudgeEntity judgeEntity = retrieveCoreObjectService.retrieveOrCreateJudge(buildFullName(judge));
```

Comment:
I have a code snippet from a Daily List that has this:-
```xml
<cs:Judiciary>
    <cs:Judge>
        <apd:CitizenNameSurname>UNUSED</apd:CitizenNameSurname>
        <apd:CitizenNameRequestedName>${JUDGE}</apd:CitizenNameRequestedName>
        <cs:CRESTjudgeID>141</cs:CRESTjudgeID>
    </cs:Judge>
</cs:Judiciary>
```

Which might indicate that we only use the requested name, so our yaml might be wrong. need to check.
i have a feeling this PR might be growing arms and legs though

Reply:
ACCEPT. you are right I am happy to change it for dailyList.